File: Java/app/src/main/java/dev/ogabek/java/activity/MainActivity.java
package dev.ogabek.java.activity;

import androidx.appcompat.app.AppCompatActivity;
import androidx.viewpager.widget.PagerAdapter;
import androidx.viewpager.widget.ViewPager;

import android.os.Bundle;

import com.google.android.material.tabs.TabLayout;

import java.util.ArrayList;
import java.util.List;

import dev.ogabek.java.R;
import dev.ogabek.java.adapter.ViewPagerAdapter;
import dev.ogabek.java.fragment.ContactFragment;
import dev.ogabek.java.fragment.PostFragment;
import dev.ogabek.java.fragment.UserFragment;
import dev.ogabek.java.model.Contact;
import dev.ogabek.java.model.Post;
import dev.ogabek.java.model.User;

public class MainActivity extends AppCompatActivity {

    private ViewPager viewPager;
    private PagerAdapter viewPagerAdapter;
    private TabLayout tabLayout;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        initViews();

    }

    private void initViews() {
        viewPager = findViewById(R.id.viewpager);
        viewPagerAdapter = new ViewPagerAdapter(getSupportFragmentManager());


        ((ViewPagerAdapter) viewPagerAdapter).add(new ContactFragment(), "Contacts");
        ((ViewPagerAdapter) viewPagerAdapter).add(new UserFragment(), "Users");
        ((ViewPagerAdapter) viewPagerAdapter).add(new PostFragment(), "Posts");


        viewPager.setAdapter(viewPagerAdapter);

        tabLayout = findViewById(R.id.tab_layout);
        tabLayout.setupWithViewPager(viewPager);
    }

    public List<Contact> prepareContactList() {
        List<Contact> contacts = new ArrayList<>();
        for (int i = 0; i < 25; i++)
            contacts.add(new Contact(R.drawable.user, "Ogabek Matyakubov", "+998 93 203 73 13"));
        return contacts;
    }

    public List<User> prepareUserList() {
        List<User> users = new ArrayList<>();
        for (int i = 0; i < 25; i++)
            users.add(new User(R.drawable.user, "Ogabek Matyakubov", "@OgabekDev"));
        return users;
    }

    public List<Post> preparePostList() {
        List<Post> posts = new ArrayList<>();
        for (int i = 0; i < 25; i++)
            posts.add(new Post(R.drawable.man, "ogabekdev", "MFaktor", R.drawable.man));
        return posts;
    }

}

File: Kotlin/app/src/main/java/dev/ogabek/kotlin/activity/MainActivity.kt
package dev.ogabek.kotlin.activity

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.viewpager.widget.PagerAdapter
import androidx.viewpager.widget.ViewPager
import com.google.android.material.tabs.TabLayout
import dev.ogabek.kotlin.R
import dev.ogabek.kotlin.adapter.ViewPagerAdapter
import dev.ogabek.kotlin.fragment.ContactListFragment
import dev.ogabek.kotlin.fragment.PostListFragment
import dev.ogabek.kotlin.fragment.UserListFragment
import dev.ogabek.kotlin.model.Contact
import dev.ogabek.kotlin.model.Post
import dev.ogabek.kotlin.model.User

class MainActivity : AppCompatActivity() {

    private lateinit var viewPager: ViewPager
    private lateinit var viewPagerAdapter: PagerAdapter
    private lateinit var tabLayout: TabLayout

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        initViews()

    }

    private fun initViews() {
        viewPager = findViewById(R.id.viewpager)
        viewPagerAdapter = ViewPagerAdapter(supportFragmentManager)


        (viewPagerAdapter as ViewPagerAdapter).add(ContactListFragment(), "Contacts")
        (viewPagerAdapter as ViewPagerAdapter).add(UserListFragment(), "Users")
        (viewPagerAdapter as ViewPagerAdapter).add(PostListFragment(), "Posts")


        viewPager.adapter = viewPagerAdapter

        tabLayout = findViewById(R.id.tab_layout)
        tabLayout.setupWithViewPager(viewPager)
    }

    fun call(number: String) {

    }

    fun prepareContactList(): List<Contact> {
        val contacts: ArrayList<Contact> = ArrayList()
        for (i in 0..25) contacts.add(Contact(R.drawable.user_48, "Ogabek Matyakubov", "+998 93 203 73 13"))
        return contacts
    }

    fun prepareUserList(): List<User> {
        val users: ArrayList<User> = ArrayList()
        for (i in 0..25) users.add(User(R.drawable.user_48, "Ogabek Matyakubov", "@OgabekDev"))
        return users
    }

    fun preparePostList(): List<Post> {
        val posts: ArrayList<Post> = ArrayList()
        for (i in 0..25) posts.add(Post(R.drawable.man, "ogabekdev", "MFaktor", R.drawable.man))
        return posts
    }

}